packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -148,7 +148,10 @@ class RenderParagraph extends RenderBox {
   }
 
   void _layoutText({ double minWidth: 0.0, double maxWidth: double.INFINITY }) {
-    _textPainter.layout(minWidth: minWidth, maxWidth: _softWrap ? maxWidth : double.INFINITY);
+    _textPainter.layout(
+        minWidth: minWidth,
+        maxWidth: _softWrap || _overflow == TextOverflow.ellipsis ? maxWidth : double.INFINITY
```

Comment:
fwiw, i'd probably use the getters rather than the fields, mostly for readability. your call.